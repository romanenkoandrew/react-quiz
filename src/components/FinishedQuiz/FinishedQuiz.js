import React from "react";
import styles from "./FinishedQuiz.module.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCheck, faTimes } from "@fortawesome/free-solid-svg-icons";
import Button from "../UI/Button/Button";
import { Link } from "react-router-dom";

const FinishedQuiz = (props) => {
  const successCount = Object.keys(props.results).reduce((total, key) => {
    if (props.results[key] === "success") {
      total++;
    }
    return total;
  }, 0);
  return (
    <div className={styles.FinishedQuiz}>
      <ul>
        {props.quiz.map((quizItem, index) => {
          const cls = `${styles.icon} ${styles[props.results[quizItem.id]]}`;
          return (
            <li key={index}>
              <strong>{index + 1}.</strong>&nbsp;
              {quizItem.question}
              {
                <FontAwesomeIcon
                  icon={
                    props.results[quizItem.id] === "error" ? faTimes : faCheck
                  }
                  className={cls}
                />
              }
            </li>
          );
        })}
      </ul>
      <p>
        Правильно {successCount} из {props.quiz.length}
      </p>
      <div>
        <Button onClick={props.onRetry} type="primary">
          Повторить
        </Button>
        <Link to="/">
          <Button type="success">Перейти в список тестов </Button>
        </Link>
      </div>
    </div>
  );
};

export default FinishedQuiz;
