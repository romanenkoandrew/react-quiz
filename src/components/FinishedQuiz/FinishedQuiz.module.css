.FinishedQuiz{
    padding: 20px;
    color: #fff;
    border: 2px solid #fff;
    border-radius: 5px;
    box-sizing: border-box;
    margin: 0 10px;
}
.FinishedQuiz ul {
    list-style: none;
    padding: 0;
    margin: 0;
}
.FinishedQuiz ul .icon {
    margin-left: 10px;
}
.FinishedQuiz ul .success {
    color: rgba(161,240,69, 0.7);
}
.success {
    color: rgba(161,240,69, 0.7)!important;
}
.error {
    color: rgba(240,87,108, 0.7) !important;
}