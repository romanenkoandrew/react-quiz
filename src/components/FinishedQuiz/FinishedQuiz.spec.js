import React from "react";
import { shallow } from "enzyme";
import FinishedQuiz from "./FinishedQuiz";

describe("FinishedQuiz component", () => {
  it("should render FinishedQuiz", () => {
    const component = shallow(
      <FinishedQuiz
        results={{ 1: "success", 2: "error" }}
        quiz={[
          { id: 1, question: "question" },
          { id: 2, question: "question2" },
        ]}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
