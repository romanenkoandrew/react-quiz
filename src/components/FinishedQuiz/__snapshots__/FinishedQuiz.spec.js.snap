// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinishedQuiz component should render FinishedQuiz 1`] = `
<div
  className="FinishedQuiz"
>
  <ul>
    <li
      key="0"
    >
      <strong>
        1
        .
      </strong>
       
      question
      <FontAwesomeIcon
        border={false}
        className="icon success"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f00c",
              "M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z",
            ],
            "iconName": "check",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </li>
    <li
      key="1"
    >
      <strong>
        2
        .
      </strong>
       
      question2
      <FontAwesomeIcon
        border={false}
        className="icon error"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              352,
              512,
              Array [],
              "f00d",
              "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
            ],
            "iconName": "times",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </li>
  </ul>
  <p>
    Правильно 
    1
     из 
    2
  </p>
  <div>
    <Button
      type="primary"
    >
      Повторить
    </Button>
    <Link
      to="/"
    >
      <Button
        type="success"
      >
        Перейти в список тестов 
      </Button>
    </Link>
  </div>
</div>
`;
